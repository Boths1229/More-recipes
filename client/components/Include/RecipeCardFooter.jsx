import React from 'react';
import EditRecipeButton from '../recipes/userRecipes/EditRecipeButton';
import EditModal from '../Include/EditModal';
import DeleteRecipe from '../recipes/userRecipes/DeleteRecipe';
import DeleteModal from '../Include/DeleteModal';

const RecipeCardFooter = (props) => {
  const {
    recipeData,
    showRecipeDetails,
    deleteRecipe,
    editRecipe,
    value,
    onChange,
    saveImageToCloud,
    handleCloseModal,
    upvoteRecipe,
    downvoteRecipe,
    favoriteRecipe,
    showActionButton,
    startEdit
  } = props.data;
  const {
    id, upvotes, downvotes, favorites
  } = recipeData;

  let component;

  if (showActionButton) {
    component = (
      <div className="card-footer card-tile">
        <div className="row">
          <div className="col-xs-6 ml-3">
            <EditRecipeButton
              recipeId={id}
              handleClick={startEdit}
              cardId={id}
              showRecipeDetails={showRecipeDetails}
            />
            <EditModal
              cardId={id}
              editRecipe={editRecipe}
              value={value}
              onChange={onChange}
              recipeId={id}
              recipeData={recipeData}
              handleCloseModal={handleCloseModal}
              saveImageToCloud={saveImageToCloud}
            />
          </div>
          <div className="col-xs-6">
            <DeleteRecipe
              recipeId={id}
            />
            <DeleteModal
              deleteRecipe={deleteRecipe}
              recipeId={id}
            />
          </div>
        </div>
      </div>
    );
  } else {
    component = (
      <ul className="list-group list-group-flush">
        <li className="list-group-item card-tile">
          <i
            className="fa fa-heart fa-2x favorite-btn"
            aria-hidden="true"
            onClick={() => favoriteRecipe(id)}
          >
            <span className="small pr-2">{favorites}
            </span>
          </i>
          <i
            className="fa fa-thumbs-up fa-2x ml-5 favorite-btn"
            aria-hidden="true"
            onClick={() => upvoteRecipe(id)}
          >
            <span className="small pr-2">{upvotes}
            </span>
          </i>
          <i
            className="fa fa-thumbs-down fa-2x favorite-btn"
            aria-hidden="true"
            onClick={() => downvoteRecipe(id)}
          >
            <span className="small pr-2">
              {downvotes}
            </span>
          </i>
        </li>
      </ul>
    );
  }
  return component;
};

export default RecipeCardFooter;
