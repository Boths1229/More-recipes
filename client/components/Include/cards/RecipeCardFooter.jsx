import React from 'react';
import EditRecipeButton from '../../Include/buttons/EditRecipeButton';
import EditModal from '../../Include/modals/EditModal';
import DeleteRecipeButton from '../../Include/buttons/DeleteRecipeButton';
import DeleteModal from '../../Include/modals/DeleteModal';

const RecipeCardFooter = (props) => {
  const {
    recipeData,
    showRecipeDetails,
    deleteRecipe,
    editRecipe,
    value,
    onChange,
    saveImageToCloud,
    handleCloseModal,
    upvoteRecipe,
    downvoteRecipe,
    favoriteRecipe,
    showActionButton,
    startEdit,
    favRecipeData
  } = props.data;
  const {
    id, upvotes, downvotes, favorites
  } = recipeData;

  let component;

  component = showActionButton ? (
    <div className="card-footer card-tile">
      <div className="row">
        <div className="col-xs-6 ml-3">
          <EditRecipeButton
            recipeId={id}
            handleClick={startEdit}
            cardId={id}
            showRecipeDetails={showRecipeDetails}
          />
          <EditModal
            cardId={id}
            editRecipe={editRecipe}
            value={value}
            onChange={onChange}
            recipeId={id}
            recipeData={recipeData}
            handleCloseModal={handleCloseModal}
            saveImageToCloud={saveImageToCloud}
          />
        </div>
        <div className="col-xs-6">
          <DeleteRecipeButton
            recipeId={id}
          />
          <DeleteModal
            deleteRecipe={deleteRecipe}
            recipeId={id}
          />
        </div>
      </div>
    </div>
  ) : component = (
    <ul className="list-group list-group-flush">
      <li className="list-group-item card-tile">
        <i
          className="fa fa-heart fa-2x favorite-btn"
          aria-hidden="true"
          onClick={() => favoriteRecipe(id)}
        >
          <span className="small pr-2">{favorites}
          </span>
        </i>
        <i
          className="fa fa-thumbs-up fa-2x ml-5 favorite-btn"
          aria-hidden="true"
          onClick={() => upvoteRecipe(id)}
        >
          <span className="small pr-2">{upvotes || favRecipeData.upvotes}
          </span>
        </i>
        <i
          className="fa fa-thumbs-down fa-2x favorite-btn"
          aria-hidden="true"
          onClick={() => downvoteRecipe(id)}
        >
          <span className="small pr-2">
            {downvotes || favRecipeData.downvotes}
          </span>
        </i>
      </li>
    </ul>
  );

  return component;
};

export default RecipeCardFooter;
