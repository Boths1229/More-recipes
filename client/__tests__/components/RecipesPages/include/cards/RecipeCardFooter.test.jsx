import React from 'react';
import { shallow } from 'enzyme';
import RecipeCardFooter from '../../../../../components/Include/cards/RecipeCardFooter';

describe('Recipe card footer component', () => {
  it('matches snapshot', () => {
    const props = {
      recipe: {}
    };
    const output = shallow(<RecipeCardFooter {...props} />);
    expect(output).toMatchSnapshot();
  });
});
