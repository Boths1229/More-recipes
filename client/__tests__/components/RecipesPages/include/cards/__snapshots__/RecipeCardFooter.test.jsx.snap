// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe card footer component matches snapshot 1`] = `
<ul
  className="list-group list-group-flush"
>
  <li
    className="list-group-item card-tile"
  >
    <i
      aria-hidden="true"
      className="fa fa-heart fa-2x btn-white"
    >
      <span
        className="small pr-2"
      />
    </i>
    <i
      aria-hidden="true"
      className="fa fa-thumbs-up fa-2x ml-5 btn-white"
    >
      <span
        className="small pr-2"
      />
    </i>
    <i
      aria-hidden="true"
      className="fa fa-thumbs-down fa-2x btn-white"
    >
      <span
        className="small pr-2"
      />
    </i>
  </li>
</ul>
`;
